File: src/main/java/org/example/kalkulationsprogramm/repository/EmailAbsenderRepository.java
package org.example.kalkulationsprogramm.repository;

import java.util.List;
import java.util.Optional;

import org.example.kalkulationsprogramm.domain.EmailAbsender;
import org.springframework.data.jpa.repository.JpaRepository;

public interface EmailAbsenderRepository extends JpaRepository<EmailAbsender, Long> {

    List<EmailAbsender> findAllByOrderBySortierungAscIdAsc();

    List<EmailAbsender> findByAktivTrueOrderBySortierungAscIdAsc();

    Optional<EmailAbsender> findFirstByAktivTrueOrderBySortierungAscIdAsc();

    Optional<EmailAbsender> findByEmailAdresseIgnoreCase(String emailAdresse);
}

File: src/main/kotlin/org/example/kalkulationsprogramm/repository/EmailAbsenderRepository.kt
package org.example.kalkulationsprogramm.repository

import java.util.Optional
import org.example.kalkulationsprogramm.domain.EmailAbsender
import org.springframework.data.jpa.repository.JpaRepository

interface EmailAbsenderRepository : JpaRepository<EmailAbsender, Long> {
    fun findAllByOrderBySortierungAscIdAsc(): List<EmailAbsender>

    fun findByAktivTrueOrderBySortierungAscIdAsc(): List<EmailAbsender>

    fun findFirstByAktivTrueOrderBySortierungAscIdAsc(): Optional<EmailAbsender>

    fun findByEmailAdresseIgnoreCase(emailAdresse: String): Optional<EmailAbsender>
}
